### ISSUE TITLE
`Task exception was never retrieved` on M1 Mac

### ISSUE THREAD

**Issue by alxmrs** [dask/distributed#7193]
**Describe the issue**:

Whenever I run a small dask distributed programs in unit tests (on my machine -- an M1 Mac), I hit an infinite loop. The same code on an Intel machine doesn't face this problem.

**Minimal Complete Verifiable Example**:

```commandline
git clone git@github.com:apache/beam.git
# create conda env for beam
cd sdks/python
pip install -e ".[dask]"
pip install pytest
pytest apache_beam/runners/dask/
```
**Anything else we need to know?**:

After I interrupt my pytest run (which is in an infinite loop), I see these traces:
```
future: <Task finished name='Task-116' coro=<Client._gather.<locals>.wait() done, defined at /opt/homebrew/Caskroom/miniconda/base/envs/beam/lib/python3.9/site-packages/distributed/client.py:2054> exception=AllExit()>
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/beam/lib/python3.9/site-packages/distributed/client.py", line 2063, in wait
    raise AllExit()
distributed.client.AllExit
Task exception was never retrieved
```

This problem seems really similar to this issue – I wonder if the solution would be a similarly small fix: 
https://github.com/microsoft/playwright-python/issues/1165

**Environment**:

- Dask version:  2022.10.0
- Python version: 3.9.13
- Operating System: M1 Mac (I think this is the core of the issue)
- Install method (conda, pip, source): `conda`.

**Comment by jrbourbeau** [comment]
Thanks for reporting @alxmrs. Do you have a code snippet you can share that reproduces the issue?

**Comment by alxmrs** [comment]
Sorry, I'm having a tough time creating a code snippet that reproduces the issue. I have provided instructions to reproduce the issue that involve running another repo's code. 

If it's easier, I'm happy to schedule a video chat to pair program on the solution.

**Comment by alxmrs** [comment]
I met with @jrbourbeau and we found that this was a problem with my installation / dependencies on my machine. The steps we took for resolution were: 
1. Move all pip dependencies into a conda environment.yml file
2. Added `conda-forge` as the only channel
3. Installed the dependencies using the `--no-default-packages`. James pointed out that there can be linking issues when mixing the default conda channel and conda-forge. 

An alternative to this setup would be to use `miniforge`, which excludes only uses the conda-forge by default.
